### PR TITLE
cask/audit: skip code signing check when the cask depends on intel

### PR DESCRIPTION
**Pull request by AkiSakurai**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

 code signing is not required on intel to run executable.

### CODE REVIEW

File: Library/Homebrew/cask/audit.rb [hunk 1]
```diff
@@ -496,7 +496,13 @@ def check_signing
           message = "Signature verification failed:\n#{result.merged_output}\nmacOS on ARM requires applications " \
                     "to be signed. Please contact the upstream developer to let them know they should "
 
-          message += if result.stderr.include?("not signed at all")
+          not_signed = result.stderr.include?("not signed at all")
+
+          depends_on_intel = cask.depends_on.arch&.any? { |arch| arch[:type] == :intel }
```
Comment (carlocab) [inline Library/Homebrew/cask/audit.rb:501]: It seems like this check should be done much earlier, no?

Comment (carlocab) [reply Library/Homebrew/cask/audit.rb:501]: Actually, no, we want this here to check for invalid signatures.

**Comment by SMillerDev** [comment]
Seeing how on x86 it's also part of the macOS security model and forces users to use `--no-quarantine` in some cases, I wouldn't recommend doing this.

**Comment by MikeMcQuaid** [comment]
> Seeing how on x86 it's also part of the macOS security model and forces users to use --no-quarantine in some cases, I wouldn't recommend doing this.

Agreed 👍🏻

**Pull request closed without merging**